src/logger/BaseLogger_test.go

```diff
@@ -0,0 +1,225 @@
+	logger1 := NewBaseLoggerAgentUrl(agent1, url1)
+	logger2 := NewBaseLoggerAgentUrl(agent2, url2)
+	logger3 := NewBaseLoggerAgentUrl(agent3, helper.demoURL)
+
+	assert.Equal(t, logger1.getAgent(), agent1)
+	assert.True(t, logger1.getEnableable())
+	assert.True(t, logger1.getEnabled())
+	assert.Equal(t, logger1.getUrl(), url1)
+
+	assert.Equal(t, logger2.getAgent(), agent2)
+	assert.True(t, logger2.getEnableable())
+	assert.True(t, logger2.getEnabled())
+	assert.Equal(t, logger2.getUrl(), url2)
+
+	assert.Equal(t, logger3.getAgent(), agent3)
+	assert.True(t, logger3.getEnableable())
+	assert.True(t, logger3.getEnabled())
+	assert.Equal(t, logger3.getUrl(), helper.demoURL)
+
+	//TODO: implement usage loggers
+	//UsageLoggers.disable();
+	//assert.False(t,UsageLoggers.GetEnabled())
+	assert.False(t,logger1.getEnabled())
+	assert.False(t,logger2.getEnabled())
+	assert.False(t,logger3.getEnabled())
+	//UsageLoggers.enable();
+	//assert.True(t,UsageLoggers.GetEnabled())
+	assert.True(t,logger1.getEnabled())
+	assert.True(t,logger2.getEnabled())
+	assert.True(t,logger3.getEnabled())
+}
+
+func TestHasValidHost(t *testing.T) {
+	host := host_lookup()
+	assert.NotNil(t, host)
+	assert.Greater(t, len(host), 0)
+	assert.Contains(t, host, "unknown")
+	assert.Equal(t, host, NewBaseLoggerAgent(MOCK_AGENT).getHost())
+}
+
+func TestHasValidVersion(t *testing.T) {
+	version := version_lookup()
+	assert.NotNil(t, version)
+	assert.Greater(t, len(version), 0)
+	//replacement of the java "startswith" assertion
+	//won't work rn since version is a dummy string
+	assert.Equal(t, strings.Index(version, "2.0."), 0)
+	assert.NotContains(t, version, "\\")
+	assert.NotContains(t, version, "\"")
+	assert.NotContains(t, version, "'")
+	assert.Equal(t, NewBaseLoggerAgent(MOCK_AGENT).getVersion(), version)
+}
+
+func TestPerformsEnablingWhenExpected(t *testing.T) {
+	helper := NewTestHelper()
+	logger := NewBaseLogger(helper.mockAgent, helper.demoURL, false, nil)
+	assert.True(t, logger.getEnableable())
+	assert.False(t, logger.getEnabled())
+	assert.Equal(t, logger.getUrl(), helper.demoURL)
+	logger.enable()
+	assert.True(t, logger.getEnabled())
+
+	queue := []string{}
+	logger = NewBaseLoggerAgentQueueEnabled(helper.mockAgent, queue, false)
+	assert.True(t, logger.getEnableable())
+	assert.False(t, logger.getEnabled())
+	assert.Nil(t, logger.getUrl())
+	logger.enable()
+	logger.disable()
+	logger.enable()
+	assert.True(t, logger.getEnabled())
+}
+
+//needs some more stuff in the baselogger class for this to compile
+func TestSkipsEnablingForInvalidUrls(t *testing.T) {
+	helper := NewTestHelper()
+	for _, url := range helper.mockURLSinvalid {
+		logger := NewBaseLogger(MOCK_AGENT, url, false, nil)
+		assert.False(t, logger.getEnableable())
+		assert.False(t, logger.getEnabled())
+		assert.Nil(t, logger.getUrl())
+		logger.enable()
+		logger.disable()
+		logger.enable()
+		assert.False(t, logger.getEnabled())
+	}
+}
+
+func TestSkipsEnablingForNullUrls(t *testing.T) {
```

Comment:
`TestSkipsEnablingForNullUrl` not `TestSkipsEnablingForNullUrls`

Reply:
Sorry I meant `TestSkipsEnablingForEmptyUrl` instead of `TestSkipsEnablingForNullUrls`